R/db-odbc-mssql.R

```diff
@@ -100,3 +100,36 @@
   )
   DBI::dbExecute(con, sql)
 }
+
+#' @export
+`db_compute.Microsoft SQL Server` <- function(
+                con,
+                table,
+                sql,
+                temporary = TRUE,
+                unique_indexes = list(),
+                indexes = list(),
+                ...) {
+
+
+  if (substr(table, 1, 1) != "#") {
+    table <- paste0("##", table)
+    message("MS SQL temp tables names require a single or double pound sign prefix. Renaming table to: ", table)
+    }
+
+  current_query <- dbplyr::sql(sql)
+  current_query <- sql_select(con, sql("*"), sql_subquery(con, current_query))
+  res <- dbGetQuery(con, current_query)
+  values <- sqlData(con, res[, , drop = FALSE])
```

Comment:
I'm probably misunderstanding the code, but is this reading the entire table into memory as a data frame? I thought the benefit of the SQL dplyr backend was that it avoided materialising the data like this?

Reply:
HI @Hong-Revo , you're right, I'm trying to remember why I made this look like `copy_to` instead of just passing the query in the FROM statement.  Let me retest and will either add another reply, or update the PR.  Thanks for reviewing it!

Reply:
I think this should do the trick. Rather than making a new PR, I'll just paste it here.
```r
#' @export
`db_compute.Microsoft SQL Server` <- function(con, table, sql, temporary=TRUE, unique_indexes=list(), indexes=list(), ...)
{
    # check that name has prefixed '##' if temporary
    if(temporary && substr(table, 1, 1) != "#")
        table <- paste0("##", table)

    if(!is.list(indexes))
        indexes <- as.list(indexes)

    if(!is.list(unique_indexes))
        unique_indexes <- as.list(unique_indexes)

    db_save_query(con, sql, table, temporary=temporary)
    db_create_indexes(con, table, unique_indexes, unique=TRUE)
    db_create_indexes(con, table, indexes, unique=FALSE)
    table
}


#' @export
`db_save_query.Microsoft SQL Server` <- function(con, sql, name, temporary=TRUE, ...)
{
    # check that name has prefixed '##' if temporary
    if(temporary && substr(name, 1, 1) != "#")
        name <- paste0("##", name)

    tt_sql <- build_sql("select * into ", ident_q(name), " from (", sql, ") ", ident_q(name), con=con)

    dbExecute(con, tt_sql)
    name
}
```

Reply:
Note: may not be Bobby Tables-resistant. Testing is advised.

Reply:
Yes, I did some testing before committing the changes

Reply:
No worries, I just spotted something else. Hope to see this on CRAN soon!